packages/dd-trace/src/telemetry/logs.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const { debugChannel, infoChannel, warnChannel, errorChannel } = require('../log/channels')
+const logCollector = require('./log_collector')
+const { sendData } = require('./send-data')
+
+const SEND_TELEMETRY = Symbol('_dd.log.SEND_TELEMETRY')
+const SEND_TELEMETRY_MARK = {
+  [SEND_TELEMETRY]: true
+}
+
+let config, application, host, interval
+
+function isSendTelemetryAvailable (messageObj) {
+  return messageObj.options &&
+    messageObj.options[SEND_TELEMETRY]
+}
+
+function addLog (messageObj, level) {
+  if (!messageObj) return
+
+  const message = messageObj.message
+
+  let logMessage
+  let stackTrace
+  if (typeof message !== 'object' || !message) {
+    logMessage = String(message)
+  } else if (!message.stack) {
+    logMessage = String(message.message || message)
+  } else {
+    logMessage = message.message || message
+    stackTrace = message.stack
+  }
+  if (stackTrace || isSendTelemetryAvailable(messageObj)) {
+    logCollector.add(logMessage, level, stackTrace)
+  }
+}
+
+function onDebug (messageObj) {
+  addLog(messageObj, 'DEBUG')
+}
+
+function onWarn (messageObj) {
+  addLog(messageObj, 'WARN')
+}
+
+function onError (messageObj) {
+  addLog(messageObj, 'ERROR')
+}
+
+function sendLogs () {
+  const logs = logCollector.drain()
+  if (logs) {
+    sendData(config, application, host, 'logs', logs)
+  }
+}
+
+function notValidConfig (config) {
+  return !config || !config.telemetry || !config.telemetry.logCollection
+}
+
+function start (aConfig, appplicationObject, hostObject, heartbeatInterval) {
```

Comment:
Can we maintain the coherence? `aConfig` for `config` and `applicationObject` for application, why not `aConfig` and `aApplication` or `configObject` and `applicationObject`?